tests/context_storages/test_dbs.py

```diff
@@ -98,25 +110,24 @@ def test_protocol_suggestion(protocol, expected):
     assert result == expected
 
 
-def test_main(testing_file, testing_context, context_id):
-    assert issubclass(DBContextStorage, DBAbstractContextStorage)
-    db = context_storage_factory(f"json://{testing_file}")
-    generic_test(db, testing_context, context_id)
-
-
 def test_shelve(testing_file, testing_context, context_id):
     db = ShelveContextStorage(f"shelve://{testing_file}")
     generic_test(db, testing_context, context_id)
+    asyncio.run(delete_shelve(db))
 
 
+@pytest.mark.skipif(not json_available, reason="JSON dependencies missing")
 def test_json(testing_file, testing_context, context_id):
-    db = JSONContextStorage(f"json://{testing_file}")
+    db = cast(JSONContextStorage, context_storage_factory(f"json://{testing_file}"))
```

Comment:
Did `cast` help increase coverage?
c8b85ad1a33d78bab78f7f6fd7d065a811583f64
If not, maybe remove it?